templates/rabbitmq.config.erb

```diff
@@ -5,9 +5,12 @@
   {ssl, [{versions, [<%= @ssl_versions.sort.map { |v| "'#{v}'" }.join(', ') %>]}]},
 <%- end -%>
   {rabbit, [
-<% if @ldap_auth -%>
+<% if @ldap_auth and (@auth_backends == 'UNSET') -%>
     {auth_backends, [rabbit_auth_backend_internal, rabbit_auth_backend_ldap]},
 <% end -%>
+<% if @auth_backends != 'UNSET' -%>
+    {auth_backends, <%= @auth_backends %>},
```

Comment:
I'd prefer that to be handled in the same way as `ssl_versions` (See line 5 above). Tthe user passes in an array and the template expands this into a properly formatted configuration entry.